### PR TITLE
feat(streaming): extend CREATE TABLE syntax to support creating append-only table source

### LINKED ISSUE #2962: RFC: extend CREATE TABLE syntax to support creating append-only table source
# RFC: extend CREATE TABLE syntax to support creating append-only table source

To support append-only table source use case and facilitate the e2e test of append-only data stream.

## Background

Currently we don't support creating an append-only table source, and to support this use case we need to extend the CREATE TABLE syntax.

## Design
The idea is to put the append-only attribute to the **_storage_parameter_** part in the WITH clause, which is consistent to CREATE SOURCE syntax.
For example, the SQL to create an append-only table named _orders_:

```
CREATE TABLE orders (
        o_orderkey BIGINT,
        o_custkey INTEGER,
        o_orderstatus VARCHAR(1),
        o_totalprice NUMERIC,
        o_orderdate DATE,
        o_orderpriority VARCHAR(15),
        o_clerk VARCHAR(15),
        o_shippriority INTEGER,
        o_comment VARCHAR(79))
        WITH (
         'appendonly'=true
        )
```
The append-only attribute will be stored in the corresponding struct in catalog. And the stream plan for MVs created on top of this table should be append-only true.

### PR DESCRIPTION
**Pull request by StrikeW**
## What's changed and what's your intention?

1. extend CREATE TABLE syntax to support creating append-only table source
2. return error when doing delete and update of an append-only table

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
close #2962

### CODE REVIEW

File: src/frontend/test_runner/tests/gen/testcases.rs [hunk 1]
```diff
@@ -0,0 +1,300 @@
+
+#[tokio::test]
```
Comment (skyzh) [inline src/frontend/test_runner/tests/gen/testcases.rs:2]: remove this file?

File: src/frontend/src/optimizer/plan_node/stream_source.rs [hunk 2]
```diff
@@ -55,9 +55,10 @@ impl fmt::Display for StreamSource {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(
             f,
-            "StreamSource {{ source: {},  columns: [{}] }}",
+            "StreamSource {{ source: {},  columns: [{}], appendonly: {} }}",
```
Comment (skyzh) [inline src/frontend/src/optimizer/plan_node/stream_source.rs:58]: only show `append only: ` when it is set to true, so as to affect as little test cases as possible

**Review by skyzh — COMMENTED** [review]
any planner test? see frontend/test_runner/README.md for more information.

File: src/frontend/src/optimizer/plan_node/stream_source.rs (hunk 2 above)
Comment (StrikeW) [reply src/frontend/src/optimizer/plan_node/stream_source.rs:58]: I grep the tests in test_runner, there is no test rely on this string. Seems fine to only affect new tests.

File: src/frontend/test_runner/tests/testdata/appendonly.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# This file is automatically generated. See `src/frontend/test_runner/README.md` for more information.
+- sql: |
+    create table t1 (v1 int, v2 int) with ('appendonly' = true);
+    select v1, max(v2) as mx2 from t1 group by v1;
+  stream_plan: |
+    StreamMaterialize { columns: [v1, agg#0(hidden), mx2], pk_columns: [v1], appendonly: true }
```
Comment (skyzh) [inline src/frontend/test_runner/tests/testdata/appendonly.yaml:6]: This derivation is wrong. HashAgg will never produce append-only result. We can fix this later.

File: src/frontend/test_runner/README.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ Given a sequence of SQL queries as the input, the test runner will check
 the logical operator tree if any, and the physical operator tree if any.
 
 The test data in YAML format is organized under `tests/testdata` folder.
+To be notice that `create materialized view` is not supported, since it is not needed.
```
Comment (skyzh) [inline src/frontend/test_runner/README.md:8]: add a blank line to separate paragraph?

**Review by skyzh — APPROVED** [review]
Rest LGTM

File: src/frontend/test_runner/tests/testdata/appendonly.yaml (hunk 3 above)
Comment (StrikeW) [reply src/frontend/test_runner/tests/testdata/appendonly.yaml:6]: Oh. you are right. The output of hash agg may produce update (U-, U+)

File: src/frontend/src/optimizer/plan_node/stream_table_scan.rs [hunk 5]
```diff
@@ -70,13 +70,19 @@ impl_plan_tree_node_for_leaf! { StreamTableScan }
...
+        builder
+            .field("table", &format_args!("{}", self.logical.table_name()))
+            .field(
+                "columns",
+                &format_args!("[{}]", self.logical.column_names().join(", ")),
+            )
+            .field("pk_indices", &format_args!("{:?}", self.base.pk_indices));
+
+        if self.append_only() {
+            builder.field("appendonly", &format_args!("{}", true));
```
Comment (fuyufjh) [inline src/frontend/src/optimizer/plan_node/stream_table_scan.rs:83]:
```suggestion
            builder.field("append_only", &format_args!("{}", true));
```

Comment (fuyufjh) [reply src/frontend/src/optimizer/plan_node/stream_table_scan.rs:83]: Idem for other operators.

**Pull request merged**